File: Phase-1-Foundations/Java/Day026_ArmstrongNumberCheck.java
package Phase_1_foundations;

import java.util.Scanner;

/**
 * Headline: Armstrong Number Check
 * Description: Check if a number equals the sum of its digits raised
 * to the power of the number of digits.
 *
 * Pseudocode:
input(num)
original = num
count = 0
sum = 0

while num != 0:
  num = num / 10
  count = count + 1

num = original

while num != 0:
  digit = num % 10
  sum = sum + Math.pow(digit, count)
  num = num / 10

if sum == original:
  print("Armstrong Number")
else:
  print("Not an Armstrong Number")
 */

// ---- Program ----
public class Day026_ArmstrongNumberCheck {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        int original = num;
        num = Math.abs(num);

        // Count digits
        int digitCount = 0;

        if (num == 0) {
            digitCount = 1;
        } else {
            int temp = num;

            while (temp > 0) {
                temp = temp / 10;
                digitCount++;
            }
        }

        // Calculate Armstrong sum
        int temp = num;
        int armstrongSum = 0;

        while (temp > 0) {
            int digit = temp % 10;
            armstrongSum = armstrongSum
                    + (int) Math.pow(digit, digitCount);

            temp = temp / 10;
        }

        // Check result
        if (original >= 0 && armstrongSum == original) {
            System.out.println("Armstrong number");
        } else {
            System.out.println("Not an Armstrong number");
        }

        sc.close();
    }
}

File: Phase-1-Foundations/Python/Day026_ArmstrongNumberCheck.py
"""
Headline: Armstrong Number Check
Description: Check if a number equals the sum of its digits raised
to the power of the number of digits.

Pseudocode:
input = int(num)
original = num
count = 0
sum = 0

while num != 0:
  num = num // 10
  count = count + 1

num = original

while num != 0:
  digit = num % 10
  sum = sum + digit ** count
  num = num // 10

if sum == original:
  print("Armstrong Number")
else:
  print("Not an Armstrong Number")
"""

# ---- Program ----
num = int(input("Enter a number: "))

original = num
num = abs(num)

# Count digits
if num == 0:
    digit_count = 1
else:
    digit_count = 0
    temp = num

    while temp > 0:
        temp = temp // 10
        digit_count = digit_count + 1

# Calculate Armstrong sum
temp = num
armstrong_sum = 0

while temp > 0:
    digit = temp % 10
    armstrong_sum = armstrong_sum + digit ** digit_count
    temp = temp // 10

# Check result
if original >= 0 and armstrong_sum == original:
    print("Armstrong number")
else:
    print("Not an Armstrong number")